### PR TITLE
Fixing broken e2e tests

### PR DESCRIPTION
**Pull request by jpbernius**
### Checklist
- [ ] I run `yarn run webpack:build:main`: the project builds without errors.
- [ ] I run `yarn lint`: the project builds without code style warnings.
- [ ] I updated the documentation and models.
- [ ] I tested the changes and all related features on the test server https://artemistest.ase.in.tum.de.
- [ ] I added (end-to-end) test cases for the new functionality.

### Motivation and Context
E2E Tests were broken, so I started working on updating the test cases to reflect the changes to the application.

### Description
Currently facing different results on Bamboo and Locally.

TODOs:
- [x] Fix Check that Quiz is answered correctly
- [x] Find out what is going on with Bamboo Results

### CODE REVIEW

**Comment by jwilland-tum** [comment]
As stated in the official mocha documentation, we should discuss avoiding arrow functions (https://mochajs.org/#arrow-functions).

**Comment by jpbernius** [comment]
Good catch @jwilland-tum.

**Comment by jwilland-tum** [comment]
Currently deletion of courses in the tests is broken due to foreign_key constraints on `Notification`s. See https://github.com/ls1intum/ArTEMiS/pull/300

**Comment by jwilland-tum** [comment]
Quiz exercise specs now work as expected. Handling Promises correctly should ensure deterministic results in different environments. Right now, test case `participate in quiz` fails for me as the view does not forward me to the solution overview after a quiz ended. If @jpbernius can confirm this faulty behaviour, I will create a bug request for that.

**Comment by jpbernius** [comment]
After discussing with @maximilianmeier, we set the `JHI_E2E_HEADLESS` Environment Variable on Bamboo, fixing the overlapping footer problem. This leaves us with on failing test during quiz participation.

File: src/main/java/de/tum/in/www1/artemis/config/SecurityConfiguration.java [hunk 1]
```diff
@@ -115,42 +111,16 @@ public void configure(WebSecurity web) {
...
-            .antMatchers("/api/files/attachments/**").permitAll()
-            .antMatchers("/api/**").authenticated()
-            .antMatchers("/websocket/tracker").hasAuthority(AuthoritiesConstants.ADMIN)
-            .antMatchers("/websocket/**").permitAll()
-            .antMatchers("/management/health").permitAll()
-            .antMatchers("/management/info").permitAll()
-            .antMatchers("/management/**").hasAuthority(AuthoritiesConstants.ADMIN)
-        .and()
-            .apply(securityConfigurerAdapter());
+        http.csrf().csrfTokenRepository(CookieCsrfTokenRepository.withHttpOnlyFalse()).ignoringAntMatchers("/websocket/**").ignoringAntMatchers("/api/lti/launch/*").and()
```
Comment (maximilianmeier) [inline src/main/java/de/tum/in/www1/artemis/config/SecurityConfiguration.java]: we should revert this

**Pull request merged**